From 2160a0f8d6889b4633d47573ac0e9d2ec965c6e0 Mon Sep 17 00:00:00 2001
From: Twin Functions <twinfunctions@outlook.com>
Date: Sat, 27 Jul 2019 18:35:56 +0800
Subject: [PATCH] scsi: use kzfree to free password

Similar to commit 34bca9bbe7a8 ("cifs: Use kzfree() to free password"),
password strings allocated for CHAP should be zero-ed before freed.

Signed-off-by: Twin Functions <twinfunctions@outlook.com>
---
 drivers/scsi/libiscsi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/scsi/libiscsi.c b/drivers/scsi/libiscsi.c
index ebd47c0cf9e9..67d66e6dbdb0 100644
--- a/drivers/scsi/libiscsi.c
+++ b/drivers/scsi/libiscsi.c
@@ -2854,8 +2854,8 @@ void iscsi_session_teardown(struct iscsi_cls_session *cls_session)
 
 	iscsi_remove_session(cls_session);
 
-	kfree(session->password);
-	kfree(session->password_in);
+	kzfree(session->password);
+	kzfree(session->password_in);
 	kfree(session->username);
 	kfree(session->username_in);
 	kfree(session->targetname);
-- 
2.11.0

